pkg/hive/hive.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hive
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	ma "github.com/multiformats/go-multiaddr"
+
+	"github.com/ethersphere/bee/pkg/hive/pb"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+const (
+	protocolName    = "hive"
+	protocolVersion = "1.0.0"
+	peersStreamName = "peers"
+	maxPO           = 7
```

Comment:
That's a good point, let's see what we do when kademlia stuff also comes by, because I have a feeling that the hive won't stay this way. Currently we are just asking for some number of peers in init, connecting all of them, and then getting notifications about new peers (not carrying for the ones in between). For now, I just need a constant number of bins here. Makes sense?

Reply:
well, `kademlia` could decide what is the max PO it would like to query. if hive takes that as a parameter then you dont need this here at all. if we take the approach of just broadcasting peers all the time - this won't be needed even in kademlia probably